source/text/labelrenderpass.ts

```diff
@@ -77,216 +77,56 @@ export class LabelRenderPass extends Initializable {
-     */
-    render3DLabels(labels: Array<Position3DLabel>): void {
-        if (labels.length === 0) {
-            log(LogLevel.Debug, `No 3D labels to render!`);
-            return;
-        }
-
-        /** We need to hold those labels because this function might be called before the font face
-         * is loaded. Once the font face is loaded, we try to render those labels again.
-         */
-
-        this._labels3D = labels;
+        const frameSize = this._camera.viewport;
+        for (const label of this._labels) {
+            label.fontFace = this._font!;
 
-        if (this._fontFace.glyphTextureExtent['0'] === 0) {
-            log(LogLevel.Debug,
-                `Will not render 3D labels as long as font face is not loaded.`);
-            return;
+            if (label instanceof Position2DLabel) {
+                glyphs2D.concat(label.typeset(frameSize));
+            } else if (label instanceof Position3DLabel) {
+                glyphs3D.concat(label.typeset());
+            }
         }
 
-        this.clear3DLabels();
-        labels.forEach((label: Position3DLabel) => {
-            this.render3DLabel(label);
-        });
+        this._geometry2D.update(glyphs2D.origins, glyphs2D.tangents, glyphs2D.ups, glyphs2D.texCoords);
+        this._geometry3D.update(glyphs3D.origins, glyphs3D.tangents, glyphs3D.ups, glyphs3D.texCoords);
     }
 
-    /**
-     * Removes all calculated vertices for 2D labels.
-     */
-    clear2DLabels(): void {
-        this._glyphs2D = new GlyphVertices(0);
-    }
 
     /**
-     * Removes all calculated vertices for 3D labels.
-     */
-    clear3DLabels(): void {
-        this._glyphs3D = new GlyphVertices(0);
-    }
-
-    /**
-     * Typesets a label and adds it to the 2D labels that will be rendered.
-     * @param label - the 2D label to be rendered.
-     */
-    render2DLabel(label: Position2DLabel): void {
-        label.fontFace = this._fontFace;
-
-        const frameSize = this._camera.viewport;
-
-        this._glyphs2D.concat(label.typeset(frameSize));
-
-        this._geometry2D.setGlyphCoords(this._glyphs2D.origins, this._glyphs2D.tangents, this._glyphs2D.ups);
-        this._geometry2D.setTexCoords(this._glyphs2D.texCoords);
-    }
-
-    /**
-     * Typesets a label and adds it to the 3D labels that will be rendered.
-     * @param label - the 3D label to be rendered.
+     * Loads a font asset and creates a FontFace
+     * @param context - Valid context to create the object for.
      */
-    render3DLabel(label: Position3DLabel): void {
-        label.fontFace = this._fontFace;
-
-        this._glyphs3D.concat(label.typeset());
+    protected loadFont(context: Context): void {
 
-        this._geometry3D.setGlyphCoords(this._glyphs3D.origins, this._glyphs3D.tangents, this._glyphs3D.ups);
-        this._geometry3D.setTexCoords(this._glyphs3D.texCoords);
+        /* This is a placeholder until the 'real' fontFace is loaded asynchronously by the fontLoader */
```

Comment:
So, the opensans is the placeholder? But this has to be loaded, too. I think the comment is outdated.